test/util.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package test
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+func AreEqualJSON(s1, s2 string) (bool, error) {
```

Comment:
Sorta related to the output reformatting discussion - I think it might be better not to have this function, and instead just do a direct `==` or `bytes.Equal()` comparison on the golden data vs. the output data. The only way that erroneously mismatched output should be able to happen there is if something in the formatting rules changes. If that happens, though, then IMO it would be expected that the golden output would have to change.